packages/@aws-cdk/aws-ecs/lib/external/external-service.ts

```diff
@@ -0,0 +1,138 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import { Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { BaseService, BaseServiceOptions, DeploymentControllerType, IBaseService, IService, LaunchType, PropagatedTagSource } from '../base/base-service';
+import { fromServiceAtrributes } from '../base/from-service-attributes';
+import { Compatibility, TaskDefinition } from '../base/task-definition';
+import { ICluster } from '../cluster';
+/**
+ * The properties for defining a service using the External launch type.
+ */
+export interface ExternalServiceProps extends BaseServiceOptions {
+  /**
+   * The task definition to use for tasks in the service.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly taskDefinition: TaskDefinition;
+
+  /**
+   * The security groups to associate with the service. If you do not specify a security group, the default security group for the VPC is used.
+   *
+   * This property is only used for tasks that use the awsvpc network mode.
+   *
+   * @default - A new security group is created.
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * Specifies whether to propagate the tags from the task definition or the service to the tasks in the service.
+   * Tags can only be propagated to the tasks within the service during service creation.
+   *
+   * @deprecated Use `propagateTags` instead.
+   * @default PropagatedTagSource.NONE
+   */
+  readonly propagateTaskTagsFrom?: PropagatedTagSource;
+}
+
+/**
+ * The interface for a service using the External launch type on an ECS cluster.
+ */
+export interface IExternalService extends IService {
+
+}
+
+/**
+ * The properties to import from the service using the External launch type.
+ */
+export interface ExternalServiceAttributes {
+  /**
+   * The cluster that hosts the service.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The service ARN.
+   *
+   * @default - either this, or {@link serviceName}, is required
+   */
+  readonly serviceArn?: string;
+
+  /**
+   * The name of the service.
+   *
+   * @default - either this, or {@link serviceArn}, is required
+   */
+  readonly serviceName?: string;
+}
+
+/**
+ * This creates a service using the External launch type on an ECS cluster.
+ *
+ * @resource AWS::ECS::Service
+ */
+export class ExternalService extends BaseService implements IExternalService {
```

Comment:
This is tricky -- if we want extend from `BaseService`, we should probably also add more validations for all the features that aren't supported. I see the validation for `CloudMapOptions` not being supported, but we'll also need to add them for the following properties/methods from [BaseService](https://docs.aws.amazon.com/cdk/api/latest/typescript/api/aws-ecs/baseservice.html#aws_ecs_BaseService):

- enableExecuteCommand
- capacityProviderSrategies
- attachToApplicationTargetGroup
- attachToClassicLB
- loadBalancerTarget
- registerLoadBalancerTargets
- configureAwsVpcNetworkingWithSecurityGroups
- autoScaleTaskCount
- networkConfiguration
- associateCloudMapOptions
- enableCloudMap

...and possibly others that I missed.

This seems like the majority of the BaseService API surface area, so we might need to ask whether or not it makes sense to extend from the BaseService class or not. If there are plans to include the things currently not supported (as enumerated in the [considerations for using ECS-A](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere.html#ecs-anywhere-considerations)), then perhaps it would. @uttarasridhar do you happen to know what the roadmap for ECS-A is?

Reply:
Thanks for pointing out the service, I have taken care of everything except `attachToClassicLB`, as this method is already deprecated.